org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/BundleUtils.java

```diff
@@ -126,22 +112,38 @@ public static void loadBundles(Collection<String> bundleLocations) throws CoreEx
 				String location = getBundleLocation(bundleLocation, true);
 
 				BundleInfo bundleInfo = getBundleInfo(bundleLocation);
-				if (!bundleInfos.add(bundleInfo)){  // returns false if bundleInfo is not added because it already exists
-					continue;
-				 }
+
 				if (bundleInfo == null) {
 					status.add(new Status(IStatus.ERROR, context.getBundle().getSymbolicName(), "Failed to get bundleInfo for bundle from " + bundleLocation, null));
 					continue;
 				}
-				Bundle bundle = Platform.getBundle(bundleInfo.getSymbolicName());
-				if (bundle != null) {
-					if (bundle.getLocation().equals(location) && bundle.getVersion().equals(Version.parseVersion(bundleInfo.getVersion()))) {
-						// Same bundle exists
+
+				Bundle[] bundles = getBundles(bundleInfo.getSymbolicName(), frameworkWiring);
+				if (bundles != null) {
+					boolean shouldSkip = false;
+					for (Bundle bundle : bundles) {
+						if (bundle.getVersion().equals(Version.parseVersion(bundleInfo.getVersion()))) {

```

Comment:
please compute Version.parseVersion(bundleInfo.getVersion())) only once, before the loop